File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/changeprogressoftask.rb
module Asposetasksjava
  module ChangeProgressOfTask
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new

        puts project.getCalculationMode()

        task = project.getRootTask().getChildren().add("Task")

        tsk = Rjb::import('com.aspose.tasks.Tsk')

        task.set(tsk.DURATION, project.getDuration(2))

        task.set(tsk.PERCENT_COMPLETE, 50)

        puts "Changed progress of task."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/ChangeProgressOfTask.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Tsk as Tsk;


class ChangeProgressOfTask{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project();

        print $project->getCalculationMode();

        $task = $project->getRootTask()->getChildren()->add("Task");

        $tsk = new Tsk();

        $task->set($tsk->DURATION, $project->getDuration(2));

        $task->set($tsk->PERCENT_COMPLETE, 50);

        print "Changed progress of task.".PHP_EOL;

    }

}
?>